tests/integration_test.rs

```diff
@@ -50,18 +50,16 @@ mod tests {
         let mut mm = get_momento_instance();
         let result = mm.create_cache(cache_name).await.unwrap_err();
         let _err_msg = "Cache name cannot be empty".to_string();
-        assert!(matches!(
-            result,
-            MomentoError::InvalidArgument(_err_message)
-        ))
+        assert!(matches!(result.to_string(), _err_message))
     }
 
     #[tokio::test]
     async fn key_id_validation() {
         let key_id = "";
         let mut mm = get_momento_instance();
         let result = mm.revoke_signing_key(key_id).await.unwrap_err();
-        assert!(matches!(result, MomentoError::InvalidArgument(_)))
+        let _expected = "".to_string();
+        assert!(matches!(result.to_string(), _expected))
     }
```

Comment:
> Adds special handling for the very very special no-error kinds of
http2 errors

What is this error case? Can we write an integ test for this to show how we're now handling it?

Reply:
I don't know how to instigate it in an integration test Vir. Server reset with no error and h2 goaway are the known causes currently. Read the documentation and/or instantiation site for the MomentoError::Interrupted enumeration for more details. If you'd like to dive deeper you can read about NO_ERROR in `h2` https://github.com/hyperium/h2/blob/07d20b19abfd3bf57bd80976089e3c24a3166bca/src/proto/connection.rs#L217-L219 and the cases in which it is the reason for `h2` family errors.